public/app/features/gops/configuration-tracker/components/Essentials.tsx

```diff
@@ -0,0 +1,188 @@
+
+import { ProgressBar, StepsStatus } from './ProgressBar';
+
+export interface EssentialsProps {
+  onClose: () => void;
+  essentialsConfig: SectionsDto;
+  stepsDone: number;
+  totalStepsToDo: number;
+}
+
+export function Essentials({ onClose, essentialsConfig, stepsDone, totalStepsToDo }: EssentialsProps) {
+  return (
+    <Drawer title="Essentials" subtitle="Complete the following configuration tasks" onClose={onClose}>
+      <EssentialContent essentialContent={essentialsConfig} stepsDone={stepsDone} totalStepsToDo={totalStepsToDo} />
+    </Drawer>
+  );
+}
+
+export function EssentialContent({
+  essentialContent,
+  stepsDone,
+  totalStepsToDo,
+}: {
+  essentialContent: SectionsDto;
+  stepsDone: number;
+  totalStepsToDo: number;
+}) {
+  return (
+    <Stack direction={'column'} gap={1}>
+      <ProgressStatus stepsDone={stepsDone} totalStepsToDo={totalStepsToDo} />
+      {essentialContent.sections.map((section: SectionDto) => (
+        <Section key={section.title} section={section} />
+      ))}
+    </Stack>
+  );
+}
+interface SectionProps {
+  section: SectionDto;
+}
+function Section({ section }: SectionProps) {
+  const styles = useStyles2(getStyles);
+  return (
+    <div className={styles.wrapper}>
+      <Text element="h4">{section.title}</Text>
+
+      <Text color="secondary">{section.description}</Text>
+      <Stack direction={'column'} gap={2}>
+        {section.steps.map((step, index) => (
+          <Step key={index} step={step} />
+        ))}
+      </Stack>
+    </div>
+  );
+}
+function DoneIcon({ done }: { done: boolean }) {
+  return done ? (
+    <Icon name="check-circle" color="green" data-testid="checked-step" />
+  ) : (
+    <Icon name="circle" data-testid="unckecked-step" />
+  );
+}
+interface StepProps {
+  step: SectionDtoStep;
+}
+
+function Step({ step }: StepProps) {
+  return (
+    <Stack direction={'row'} justifyContent={'space-between'} data-testid="step">
+      <Stack direction={'row'} alignItems="center">
+        {step.button.done !== undefined && <DoneIcon done={step.button.done} />}
+        <Text variant="body">{step.title}</Text>
+        <Tooltip content={step.description} placement="right">
+          <Icon name="question-circle" />
+        </Tooltip>
+      </Stack>
+      <StepButton {...step.button} data-testid="step-button" />
+    </Stack>
+  );
+}
+
+interface LinkButtonProps {
+  urlLink?: { url: string; queryParams?: Record<string, string> };
+  label: string;
+  urlLinkOnDone?: { url: string; queryParams?: Record<string, string> };
+  labelOnDone?: string;
+  done?: boolean;
+}
+function OpenLinkButton(props: LinkButtonProps) {
+  const { urlLink, label, urlLinkOnDone, labelOnDone, done } = props;
+  const urlToGoWhenNotDone = urlLink?.url
+    ? createUrl(urlLink.url, {
+        returnTo: location.pathname + location.search,
+        ...urlLink.queryParams,
+      })
+    : '';
+  const urlToGoWhenDone = urlLinkOnDone?.url
```

Comment:
I think this is incorrect, the base URL should still be `urlLink` and not `urlLinkOnDone` right?

Reply:
in this case we need to use `urlLinkOnDone` as we are creating the link for the 'task done' use case.